swiftpkg/internal/swift_package_tool_runner_template.sh

```diff
@@ -43,12 +47,26 @@ else
   args+=("--disable-dependency-cache")
 fi
 
+if [ "$replace_scm_with_registry" = "true" ]; then
+  args+=("--replace-scm-with-registry")
+fi
+
+if [ "$use_registry_identity_for_scm" = "true" ]; then
+  args+=("--use-registry-identity-for-scm")
+fi
+
 args+=("--manifest-cache=$manifest_cache")
 
+# If registries_json is set, symlink the `.json` file to the `config_path/configuration` directory.
+if [ -n "$registries_json" ]; then
+  mkdir -p "$config_path" && ln -sf "$(readlink -f "$registries_json")" "$config_path/registries.json"
+fi
```

Comment:
> Won't this blow away a real registries.json file if one exists?

No, this is symlinking the real file into the rules runfiles so that SPM uses that one from the rules `--config-path`

Reply:
> I think you're also missing the configuration directory in here.

Seems to work without one but i'll update to follow the normal `.swiftpm` layout

Reply:
Actually, seems like `--config-path` defines the `configuration` directory path, it's just that by default it's nested within `.swiftpm`. Going to keep this as-is but rename the default `config_path` value to `config` so its not confusing